components/script/dom/htmlcollection.rs

```diff
@@ -205,24 +205,24 @@ impl HTMLCollection {
         HTMLCollection::create(window, root, box ElementChildFilter)
     }
 
-    pub fn elements_iter_after(&self, after: &Node) -> HTMLCollectionElementsIter {
+    pub fn elements_iter_after<'a>(&'a self, after: &'a Node) -> impl Iterator<Item=Root<Element>> + 'a {
         // Iterate forwards from a node.
         HTMLCollectionElementsIter {
-            node_iter: box after.following_nodes(&self.root),
+            node_iter: after.following_nodes(&self.root),
             root: Root::from_ref(&self.root),
             filter: &self.filter,
         }
     }
 
-    pub fn elements_iter(&self) -> HTMLCollectionElementsIter {
+    pub fn elements_iter<'a>(&'a self) -> impl Iterator<Item=Root<Element>> + 'a {
```

Comment:
Result of your proposed changes:

```
> ./mach build -d
   Compiling script v0.0.1 (file:///Users/coreyf/Development/rust/servo/components/script)
error[E0495]: cannot infer an appropriate lifetime for borrow expression due to conflicting requirements
   --> /Users/coreyf/Development/rust/servo/components/script/dom/htmlcollection.rs:213:21
    |
213 |             filter: &self.filter,
    |                     ^^^^^^^^^^^^
    |
note: first, the lifetime cannot outlive the anonymous lifetime #1 defined on the block at 208:99...
   --> /Users/coreyf/Development/rust/servo/components/script/dom/htmlcollection.rs:208:100
    |
208 |     pub fn elements_iter_after(&self, after: &Node) -> impl Iterator<Item=Root<Element>> + 'static {
    |                                                                                                    ^
note: ...so that reference does not outlive borrowed content
   --> /Users/coreyf/Development/rust/servo/components/script/dom/htmlcollection.rs:213:21
    |
213 |             filter: &self.filter,
    |                     ^^^^^^^^^^^^
    = note: but, the lifetime must be valid for the static lifetime...
note: ...so that the type `dom::htmlcollection::HTMLCollectionElementsIter<'_, dom::node::FollowingNodeIterator>` will meet its required lifetime bounds
   --> /Users/coreyf/Development/rust/servo/components/script/dom/htmlcollection.rs:208:56
    |
208 |     pub fn elements_iter_after(&self, after: &Node) -> impl Iterator<Item=Root<Element>> + 'static {
    |                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0495]: cannot infer an appropriate lifetime for borrow expression due to conflicting requirements
   --> /Users/coreyf/Development/rust/servo/components/script/dom/htmlcollection.rs:227:21
    |
227 |             filter: &self.filter,
    |                     ^^^^^^^^^^^^
    |
note: first, the lifetime cannot outlive the anonymous lifetime #1 defined on the block at 222:101...
   --> /Users/coreyf/Development/rust/servo/components/script/dom/htmlcollection.rs:222:102
    |
222 |     pub fn elements_iter_before(&self, before: &Node) -> impl Iterator<Item=Root<Element>> + 'static {
    |                                                                                                      ^
note: ...so that reference does not outlive borrowed content
   --> /Users/coreyf/Development/rust/servo/components/script/dom/htmlcollection.rs:227:21
    |
227 |             filter: &self.filter,
    |                     ^^^^^^^^^^^^
    = note: but, the lifetime must be valid for the static lifetime...
note: ...so that the type `dom::htmlcollection::HTMLCollectionElementsIter<'_, dom::node::PrecedingNodeIterator>` will meet its required lifetime bounds
   --> /Users/coreyf/Development/rust/servo/components/script/dom/htmlcollection.rs:222:58
    |
222 |     pub fn elements_iter_before(&self, before: &Node) -> impl Iterator<Item=Root<Element>> + 'static {
    |                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```